WordPress/src/main/java/org/wordpress/android/util/PerAppLocaleManager.kt

```diff
@@ -0,0 +1,132 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.provider.Settings
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.core.os.LocaleListCompat
+import org.wordpress.android.fluxc.utils.AppLogWrapper
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import java.util.Locale
+import javax.inject.Inject
+
+/**
+ * Helper class to manage AndroidX per-app language preferences
+ * https://developer.android.com/guide/topics/resources/app-languages
+ */
+class PerAppLocaleManager @Inject constructor(
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val appLogWrapper: AppLogWrapper,
+) {
+    private fun getCurrentLocale(): Locale {
+        return if (isApplicationLocaleEmpty()) {
+            Locale.getDefault()
+        } else {
+            getApplicationLocaleList()[0] ?: Locale.getDefault()
+        }
+    }
+
+    fun getCurrentLocaleDisplayName(): String = getCurrentLocale().displayName
+
+    /**
+     * Important: this should only be called after Activity.onCreate()
+     * https://developer.android.com/reference/androidx/appcompat/app/AppCompatDelegate#getApplicationLocales()
+     */
+    private fun getApplicationLocaleList() = AppCompatDelegate.getApplicationLocales()
+
+    private fun isApplicationLocaleEmpty(): Boolean {
+        val locales = getApplicationLocaleList()
+        return (locales.isEmpty || locales == LocaleListCompat.getEmptyLocaleList())
+    }
+
+    /**
+     * We want to make sure the language pref for the in-app locale (old implementation) is set
+     * to the same locale as the AndroidX per-app locale. This way LocaleManager.getLanguage -
+     * which is used throughout the app - returns the correct language code. We can remove
+     * this once the per-app language pref is no longer experimental.
+     */
+    private fun checkOldLanguagePrefKey() {
+        val prefKey = LocaleManager.getLocalePrefKeyString()
+        val inAppLanguage = appPrefsWrapper.prefs().getString(prefKey, "")
```

Comment:
I'm not sure this is possible without making more changes, because `AppPrefs.getString` expects a `PrefKey` rather than a string:

```
private static String getString(PrefKey key, String defaultValue)
```

Reply:
Ah, apologies for not being clear on that, what I meant was for you to remove the exposure of this `pref()` function from the `AppsPrefsWrapper` class:

```kotlin
    fun prefs(): SharedPreferences {
        return AppPrefs.prefs()
    }
```

And instead move the implementation details for `prefs` related `get/putString(...)` into it, something like:

```kotlin
    fun prefsGetString(key: String, defValue: String): String? {
        return AppPrefs.prefs().getString(key, defValue)
    }

    fun prefsPutString(key: String, defValue: String) {
        AppPrefs.prefs().edit().putString(key, defValue).apply()
    }
```

Then, every client that needs such functionality, like `PerAppLocaleManager.checkAndUpdatedOldLanguagePrefKey()`, would just do the below, thus hiding the implementation details, moving that logic into the wrapper:

```kotlin
appPrefsWrapper.prefsGetString(prefKey, "")
appPrefsWrapper.prefsPutString(prefKey, perAppLanguage)
```

If that doesn't make sense to you, please ignore... 😊 🙏

Reply:
Ah, got it! I took your advice in [84a5a9d](https://github.com/wordpress-mobile/WordPress-Android/pull/21504/commits/84a5a9de05c854c912f9598468040d564785da52).

Reply:
Awesome, thanks Nick! ❤️